src/components/Heading/Heading.tsx

```diff
@@ -1,56 +1,82 @@
 import clsx from 'clsx';
-import React, { ReactNode } from 'react';
+import React, { forwardRef } from 'react';
 import styles from './Heading.module.css';
 
-export interface Props {
+export const VARIANTS = [
+  'alert',
+  'base',
+  'brand',
+  'inherit',
+  'neutral',
+  'success',
+  'warning',
+  'white',
   /**
-   * Child node(s) that can be nested inside component
-   */
-  children: ReactNode;
+   * @deprecated Info variant is deprecated.
+   */ 'info',
+] as const;
+export type Variant = typeof VARIANTS[number];
+export type HeadingElement = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+
+type Props = {
   /**
-   * CSS class names that can be appended to the component.
+   * This prop can be used to specify which size heading should
+   * actually be rendered, in the case that you want to render an element
+   * as one heading but style it as if it were another. If both an `as` prop
+   * and a `size` prop are passed, the `as` will be used to determine the element
+   * and the `size` will be used to determine the styling.
    */
+  as?: HeadingElement;
+  children: React.ReactNode;
   className?: string;
-  /**
-   * Id of the heading
-   */
-  id?: string;
-  /**
-   * Size variants of the Heading component, ranging from 1-7. These numbers map to typographic sizing/style presets. This prop is used in instances where a Heading needs to use a specific tag (such as `h3`) for semantic reasons, but needs to be styled like something other than its default styles (such as an `h2` or `h1`).
-   */
-  size?: 1 | 2 | 3 | 4 | 5 | 6;
```

Comment:
I used same typing for `size` as `as`. As used in EDS v0 and I feel would be less confusing